### ISSUE TITLE
Multiple Bots refreshing causing 401 Unauthorized

### ISSUE THREAD

**Issue by virtualxdriver** [OpenWonderLabs/homebridge-switchbot-openapi#42]
**Describe The Bug:**

Since a few hours ago, all API calls are failing with 401 Unauthorized.
I tried to reset my token but calls still fail with the new one.

**To Reproduce:**

Try to toggle a bot via Homebridge. The call fails but the bot’s status still changes in the app.

**Expected behavior:**

The call should go through. If it fails, the button in HomeKit should clearly display that an error happened.

**Logs:**

```
Failed to Discover Devices. `Request failed with status code 401.`
```

**Homebridge Config:**

```json
Paste your homebridge config.json here.
Remove any sensitive information, such as your homebridge-switchbot-openapi token.

***Read and Paste over this Text the Text Above^^^^^^
```

**Screenshots:**

**Device Model:**

# **Environment:**

- **Node.js Version**: 
- **NPM Version**: 
- **Homebridge Version**: 
- **Homebridge SwitchBot Plugin Version**:
- **Homebridge Config UI X Plugin Version**:
- **Operating System**:

**Comment by donavanbecker** [comment]
@virtualxdriver If you would have filled in all info requested with this template it would have never closed this automatically.

**Comment by virtualxdriver** [comment]
Pretty standard configuration with 6 bots on the same account. Everything was working fine until a few hours ago.
I suspect the issue is more a temporary blip on the SwitchBot side so please feel free to close this.

However, I do believe #43 is valid: errors should be handled so the user can notice something is wrong.
I only noticed something was wrong when I came home to a cold apartment because my heater did not start. 😅

**Comment by donavanbecker** [comment]
> Pretty standard configuration with 6 bots and a hub mini on the same account. Everything was working fine until a few hours ago.
> I suspect the issue is more a temporary blip on the SwitchBot side so please feel free to close this.
> 
> However, I do believe #43 is valid: errors should be handled so the user can notice something is wrong.
> I only noticed something was wrong when I came home to a cold apartment because my heater did not start. 😅

if you can post your config, it will give me a better idea. usually 401 is to many request.

**Comment by virtualxdriver** [comment]
Sure, here you are:
```
{
            "name": "SwitchBot",
            "credentials": {
                "openToken": "[REDACTED]",
                "notice": "Keep your token a secret!"
            },
            "options": {
                "bot": {
                    "device_switch": [
                        "F1D5541FD11E",
                        "EFC281C1DF11",
                        "D7A225A579D8",
                        "F11E64D9E15A",
                        "E02933CF1F2C"
                    ],
                    "device_press": [
                        "E62D26A9EE8F"
                    ]
                }
            },
            "devicediscovery": false,
            "platform": "SwitchBot"
        }
```

Thanks for the great plugin! 👍

**Comment by donavanbecker** [comment]
@virtualxdriver since you don't have it set in the config. can you set your `refreshRate`? I would set it to like `400` and see if that helps. the more devices you have the more it talks to the api, which is why you are getting a 401 error.

Once you set your `refreshRate` you will have to wait till the api pans out and allows you to connect again. 

See API [Request Limit](https://github.com/OpenWonderLabs/SwitchBotAPI#request-limit).

**Comment by virtualxdriver** [comment]
OK, I’ve set it to 600 for now and will report.

What’s even the point of polling the API at all for bots? I can understand for a Meter, but Bots don’t report status so it feels unnecessary.

**Comment by donavanbecker** [comment]
> OK, I’ve set it to 600 for now and will report.
> 
> What’s even the point of polling the API at all for bots? I can understand for a Meter, but Bots don’t report status so it feels unnecessary.

yeah you have a good point, only reason I was doing it for bots was to make sure they they were still connected with the status, but not even sure if that shows the status of the bot or just the API connection. I will remove it in the next update!

**Comment by donavanbecker** [comment]
@virtualxdriver I have removed the status of the Bots, from being pulled, in the latest Beta. So this should not cause issues with the refreshRate.

So if you could try the latest beta and give me feedback let me know.
Beta: v1.2.5-beta.2

**Comment by virtualxdriver** [comment]
Thanks! I have just installed the new beta and reset the refresh rate setting. Will let you know how it goes!

**Comment by virtualxdriver** [comment]
With the 1.2.5 beta, I noticed that bots set up in press mode do not automatically reset back to off when pressed. Could it be a regression introduced by the change?

**Comment by donavanbecker** [comment]
> With the 1.2.5 beta, I noticed that bots set up in press mode do not automatically reset back to off when pressed. Could it be a regression introduced by the change?

Yeah I notice that right before going to bed. I will work on getting that fixed! Should be a big issue since on or off will cause the bot to press, but I like that it just turns it off right away.

**Comment by donavanbecker** [comment]
> With the 1.2.5 beta, I noticed that bots set up in press mode do not automatically reset back to off when pressed. Could it be a regression introduced by the change?

Okay I have corrected this!

So if you could try the latest beta and give me feedback let me know.
Beta: v1.2.5-beta.3

**Comment by virtualxdriver** [comment]
After living on it for a few days, I can confirm the issue is addressed with 2.0.0. Thanks!

**Comment by donavanbecker** [comment]
> After living on it for a few days, I can confirm the issue is addressed with 2.0.0. Thanks!

Awesome! Thanks for letting me know. And sorry for having to release v2.0.0, need to release it sooner then later so we don't run into duplicate devices, if changed between devices.
